### PR TITLE
Use correct encoding for temp diff files.

### LINKED ISSUE #1137: Pull Request shows as modified even when it isn't
The inline reviews feature considers the `boo` file in the following PR as modified even when it isn't:

https://github.com/tierninho-tester/dasdasdasd/pull/4

This appears to be because the `boo` file has no BOM but visual studio adds one in the editor. The editor contents are then passed to `GitClient.IsModified` via the `contents` parameter where they are compared with the file in the repository, and git considers them different because of the mismatched BOM.

### PR DESCRIPTION
**Pull request by grokys**
The encoding logic in `PullRequestDetailView` was completely wrong - it was extracting the right file with UTF8 encoding and then using that for the left file. Moved the encoding detection logic to `PullRequestDetailViewModel`.

Fixes #1137

### CODE REVIEW

File: src/GitHub.InlineReviews/Services/PullRequestSessionService.cs [hunk 1]
```diff
@@ -61,9 +61,10 @@ class PullRequestSessionService : IPullRequestSessionService
         public async Task<bool> IsUnmodifiedAndPushed(ILocalRepositoryModel repository, string relativePath, byte[] contents)
         {
             var repo = await GetRepository(repository);
+            var modified = await gitClient.IsModified(repo, relativePath, contents);
+            var pushed = await gitClient.IsHeadPushed(repo);
```
Comment (grokys) [inline src/GitHub.InlineReviews/Services/PullRequestSessionService.cs]: Moved these checks to separate variables like this for ease of debugging.

File: src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs [hunk 2]
```diff
@@ -477,20 +477,14 @@ public override void Initialize(ViewWithData data)
         /// </param>
         /// <param name="encoding">The encoding to use.</param>
```
Comment (jcansdale) [inline src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs]: This should be `head` not `encoding`.

Comment (grokys) [inline src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs]: The comment for `head` is already there. It's just `encoding` needs removing.

**Comment by jcansdale** [comment]
We're currently special casing UTF8, but I think this is fine because UTF8 is the only text format with a signature that VS currently supports.

Here is the `Advanced Save Options...`.
image

**Pull request merged**